### DISCUSSION TITLE
API `variables/{variable_key}` request fails if key has character `/`

### DISCUSSION THREAD

**Discussion by jcom000** [apache/airflow#25122] [General]
### Apache Airflow version

2.3.2

### What happened

Created a variable e.g. `a/variable` and couldn't get or delete it

### What you think should happen instead

i shouldn't've been allowed to create a variable with `/`, or the GET and DELETE should work

### How to reproduce

![image](https://user-images.githubusercontent.com/98349137/179311482-e3a74683-d855-4013-b27a-01dfae7db0ff.png)
![image](https://user-images.githubusercontent.com/98349137/179311639-7640b0b5-38c6-4002-a04c-5bcd8f8a0784.png)

```
DELETE /variables/{variable_key}
GET /variables/{variable_key}
```
create a variable with `/`, and then try and get it. the get will 404, even after html escape. delete also fails
`GET /variables/` works just fine

### Operating System

astro

### Deployment

Astronomer

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by mik-laj** [comment]
You need to url-encode this parametr i e replace `/` to `%2F`.  Can you check if everything works then?

**Comment by potiuk** [comment]
Converting it into a discussion - it's not clear if there is another solution here. We know using "/" in dag id is problematic in a number of cases, but it's not clear if it is an issue that needs documenting, explanation, prevention, or maybe leaving as is.

**Comment by jedcunningham** [comment]
It wasn't an encoding problem, rather the 'string' flask converter doesn't accept slashes leading to the url not matching a route. Using 'path' allows it to accept slashes as well.

#25774 will fix this.
